.github/workflows/auto-approve-merge.yml

```diff
@@ -0,0 +1,27 @@
+# Basic workflow that approves and enables auto merge for PRs.
+# Currently it is used to automate merging dependabot PRs or overridining the required review on a PR
+# https://app.getguru.com/card/cbjEdoAi/Dependabot-and-auto-merging
+
+name: Enable auto merge and then approve
+
+on: workflow_call
+
+jobs:
+  auto-approve-merge-dependency-updates:
+    name: Approve and Enable auto merge on dependabot PRs
+    runs-on: ubuntu-latest
+    steps:
+    # Use github cli to enable auto approve. Only works on repos which have auto merge enabled and branch protections setup which is currently blocking a merge
+    # for example still requires 1 review or a required ci step has not succeeded.
+    - if: github.actor == 'dependabot[bot]'
+      name: Enable Automerge
+      run: gh pr merge --auto --merge "$PR_URL"
+      env:
+        PR_URL: ${{ github.event.pull_request.html_url }}
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    # v2.1.0 as of 02/23/2022, reviewed by eng-security.
+    - if: github.actor == 'dependabot[bot]' || contains( github.event.pull_request.labels.*.name, 'auto approve')
```

Comment:
Is this safe to use on public repos? For example, do non-Clever folks have access to add labels?

Reply:
and/or can a malicious actor impersonate Dependabot.

Reply:
- non-Clever folks should not have access to add labels. You need to have at least "triage" level permissions to apply labels but non-clever folks should just have read permissions.

- Not sure about if someone can impersonate the github actor. I can investigate more on this. Maybe it makes sense to replace it with `github.event.pull_request.user.login`

Reply:
`github.actor` is user that initiated the workflow run whereas `github.event.pull_request.user.login` is user that created the pull request

Reply:
I am reading https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#understanding-the-risk-of-script-injections and looks like `github.actor` can be trusted. It is stuff like `github.event.issue.title` or `github.event.pull_request.body` where we need to be careful as that can be edited by the any user.